### PR TITLE
Add support for openIdConnect auth scheme

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix [#2774](https://github.com/microsoft/typespec/issues/2774)

### CODE REVIEW

File: common/changes/@typespec/http/openapi3-openid_2024-01-23-16-56.json
Comment (timotheeguerin) [inline common/changes/@typespec/http/openapi3-openid_2024-01-23-16-56.json:1]: remove rush changelogs

**Review by markcowl — CHANGES_REQUESTED** [review]
Some minor nits on docs

File: eng/pipelines/jobs/publish-npm.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+parameters:
```
[the remark this answers is not in the document]
Comment (timotheeguerin) [reply eng/pipelines/jobs/publish-npm.yml:1]: ah oops

File: docs/libraries/http/reference/data-types.md [hunk 2]
```diff
@@ -224,6 +224,23 @@ The request has succeeded.
 model TypeSpec.Http.OkResponse
 ```
 
+### `OpenIdConnectAuth` {#TypeSpec.Http.OpenIdConnectAuth}
+
+OpenID Connect (OIDC) is an identity layer built on top of the OAuth 2.0 protocol and supported by some OAuth 2.0 providers, such as Google and Azure Active Directory.
+It defines a sign-in flow that enables a client application to authenticate a user, and to obtain information (or "claims") about that user, such as the user name, email, and so on.
+User identity information is encoded in a secure JSON Web Token (JWT), called ID token.
+OpenID Connect defines a discovery mechanism, called OpenID Connect Discovery, where an OpenID server publishes its metadata at a well-known URL, typically
```
[the remark this answers is not in the document]
Comment (timotheeguerin) [reply docs/libraries/http/reference/data-types.md:232]: I'm not sure, copied the openapi3 documentation for this

[the remark this answers is not in the document]
Comment (timotheeguerin) [reply docs/libraries/http/reference/data-types.md:232]: oh... Yeah missed the example

**Review by markcowl — APPROVED** [review]

**Pull request merged**